### PR TITLE
feat: example job to manane webhooks

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It add a example of how to manage a webhook with `generic-webhook-trigger`

## Why is it important?

It can process the webhook payload into a job to implement functionality we do not have in Multibranch pipelines.

### CODE REVIEW

File: resources/webhook-template.groovy [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      [key: 'after', value: '$.after'],
+      [key: 'payload', value: '$'],
+      [key: 'comment_id', value: '$.comment.id'],
+      [key: 'payload', value: '$'],
+     ],
+    genericHeaderVariables: [
+     [key: 'x-github-event', regexpFilter: '']
+    ],
+     causeString: 'Triggered on $ref',
+     //tokenCredentialId: '',
```
Comment (cachedout) [inline resources/webhook-template.groovy:56]:
```suggestion
```
Nit: removed unused code

File: resources/webhook-template.groovy [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     printContributedVariables: true,
+     printPostContent: false,
+     silentResponse: true,
+     regexpFilterText: '$ref-$x_github_event',
+     //regexpFilterExpression: '^(refs/tags/current|refs/heads/master/.+)-push$'
```
Comment (cachedout) [inline resources/webhook-template.groovy:61]:
```suggestion
```
Nit: remove unused code

**Review by cachedout — APPROVED** [review]

File: resources/webhook-template.groovy (hunk 1 above)
Comment (kuisathaverat) [inline resources/webhook-template.groovy:56]:
```suggestion
     //Allow to use a credential as a secret to trigger the webhook
     //tokenCredentialId: '',
```

File: resources/webhook-template.groovy (hunk 2 above)
Comment (kuisathaverat) [inline resources/webhook-template.groovy:61]:
```suggestion
     regexpFilterExpression: '^(refs/tags/current|refs/heads/master/.+)-comment$'
```

File: resources/webhook-template.groovy [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  // triggers {
+  //   // most of then come from https://prow.k8s.io/command-help
+  //   issueCommentTrigger('(?i)^\\/(run|test|lgtm|cc|assign|approve|meow|woof|bark|this-is-|lint|help|hold|label|close|reopen|skip|ok-to-test|package|build|deploy)(-\\w+)?(\\s+.*)?$')
+  // }
```
Comment (kuisathaverat) [inline resources/webhook-template.groovy:40]:
```suggestion
```

File: resources/webhook-template.groovy (hunk 1 above)
Comment (kuisathaverat) [reply resources/webhook-template.groovy:53]: I've added a comment on top of it

**Pull request merged**